internal/integtest/setup.go

```diff
@@ -108,12 +113,13 @@ func HNCBeforeSuite() {
 	Expect(K8sClient).ToNot(BeNil())
 
 	go func() {
-		err = k8sManager.Start(ctrl.SetupSignalHandler())
```

Comment:
Hmm, sudden thought. Instead of creating the cancelable context from `context.Background()`, should it be from `ctrl.SetupSignalHandler()` instead? What are we losing by not using this?

Reply:
Good idea - if it works. I can see if that works in my environment.

Reply:
PTAL

Reply:
I am uncertain if my latest change (moving the initialization of the cancel-func) is a good idea. Ginkgo seems to want to run AfterSuite not matter what happens. And the cancel-func might be uninitialized (`nil`).

Using `ctrl.SetupSignalHandler()` instead of creating a new context works!

Reply:
What about checking whether `k8sManagerCancelFn` is nil or not before calling it in `HNCAfterSuite`? Otherwise this lgtm.

Reply:
Done